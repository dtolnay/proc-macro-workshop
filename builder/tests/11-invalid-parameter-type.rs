// Ensure that your macro reports a reasonable error message when the caller
// mistypes the inert attribute in various ways. This is a compile_fail test.
//
// The preferred way to report an error from a procedural macro is by including
// an invocation of the standard library's compile_error macro in the code
// emitted by the procedural macro.
//
//
// Resources:
//
//   - The compile_error macro for emitting basic custom errors:
//     https://doc.rust-lang.org/std/macro.compile_error.html
//
//   - Lowering a syn::Error into an invocation of compile_error:
//     https://docs.rs/syn/0.15/syn/struct.Error.html#method.to_compile_error

use derive_builder::Builder;

#[derive(Builder)]
pub struct Command {
    executable: String,
    #[builder]
    args: Vec<String>,
    env: Vec<String>,
    current_dir: Option<String>,
}

fn main() {}
