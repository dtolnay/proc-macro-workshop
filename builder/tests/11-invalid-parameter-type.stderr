error: expected builder attribute to be a list
   |
22 |     #[builder]
   |     ^^^^^^^^^^
